### PR TITLE
postgres: Add `character` data type

### PR DESCRIPTION
**Pull request by emk**
This is simple case: We just need to add a new grammar rule to the SQL
parser, and to the code that parses `information_schema.columns`. From
there, we can safely translate it to `text` immediately, and the tests
all pass. No need to change the Postgres BINARY encoder, because
apparently `character` is encoded the same way as `varchar` and `text`
on the wire.

This has not been tested with Redshift, but if a given PostgreSQL type
works at all with Redshift, then we usually don't need to do more than
this.

@wboykinm Do the test `*.csv` and `*.sql` files look good? In the future we can just add new columns to these files.

@fwallacevt If you hear about another one of these errors, and you want to take a shot at it, you can probably just prepare a PR like this one updating the existing test case.

### CODE REVIEW

**Comment by rossmeissl** [comment]
@emk I love reading your PRs

File: dbcrossbar/fixtures/more_pg_types.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+CREATE TABLE more_pg_types (
+    c1 character,
+    c2 character(2),
+    c3 character (3)
```
Comment (wboykinm) [inline dbcrossbar/fixtures/more_pg_types.sql]: @emk Is the space here between `character` and `(3)` a deliberate syntax variation test?

File: dbcrossbarlib/src/drivers/postgres_shared/catalog.rs [hunk 2]
```diff
@@ -200,6 +200,7 @@ fn pg_data_type(
         let ty = match data_type {
             "bigint" => Ok(PgScalarDataType::Bigint),
             "boolean" => Ok(PgScalarDataType::Boolean),
+            "character" => Ok(PgScalarDataType::Text),
```
Comment (wboykinm) [inline dbcrossbarlib/src/drivers/postgres_shared/catalog.rs]: I love the implication here: "FFS, it's all just text"

File: dbcrossbar/fixtures/more_pg_types.sql (hunk 1 above)
Comment (emk) [reply dbcrossbar/fixtures/more_pg_types.sql]: @wboykinm Yeah, I like to mix things up a bit in the test cases sometimes. No point in testing the same thing twice. :-)

File: dbcrossbarlib/src/drivers/postgres_shared/catalog.rs (hunk 2 above)
Comment (emk) [reply dbcrossbarlib/src/drivers/postgres_shared/catalog.rs]: Yeah, basically we're choosing to live in denial unless it makes a difference, like it does for `json` and `jsonb`, which need separate handing in the PostgreSQL BINARY format.

`dbcrossbar` is one giant exercise in compromise and negotiation. It's the only way to get something like this off the ground gradually.

**Pull request merged**